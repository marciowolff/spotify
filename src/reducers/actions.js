export { default as actionsSpotify } from './spotify/actions';
