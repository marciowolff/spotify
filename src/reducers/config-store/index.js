import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../';

export default ({ initialStates } = {}) => {
  const enhancer = compose(
    applyMiddleware(thunk),
    logger()
  );

  return createStore(rootReducer, initialStates, enhancer);
};

const logger = () =>
  window.__REDUX_DEVTOOLS_EXTENSION__
    ? window.__REDUX_DEVTOOLS_EXTENSION__()
    : x => x;
