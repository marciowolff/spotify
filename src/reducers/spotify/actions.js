export const SPOTIFY_LIST = 'spotify:SPOTIFY_LIST';
export const SPOTIFY_ALBUM = 'spotify:SPOTIFY_ALBUM';

const setList = (albums = []) => dispatch => {
  dispatch({
    type: SPOTIFY_LIST,
    payload: albums,
  });
};

const setAlbum = (content = {}) => dispatch => {
  dispatch({
    type: SPOTIFY_ALBUM,
    payload: content,
  });
};

export default {
  setList,
  setAlbum,
};
