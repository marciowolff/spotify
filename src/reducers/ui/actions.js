export const ALERT_MESSAGE = 'ui:ALERT_MESSAGE';
export const LOADING = 'ui.LOADING';

export const alertMessage = (message = '') => dispatch => {
  dispatch({
    type: ALERT_MESSAGE,
    payload: {
      message,
    },
  });
};

export const setLoading = (loading = false) => dispatch => {
  dispatch({
    type: LOADING,
    payload: {
      loading,
    },
  });
};
