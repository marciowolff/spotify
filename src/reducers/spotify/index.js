import createReducer from '../create-reducer';

import { SPOTIFY_LIST, SPOTIFY_ALBUM } from './actions';

export const initialState = {};

const spotify = createReducer(initialState, {
  [SPOTIFY_LIST]: (state, action) => ({
    ...state,
    list: action.payload,
  }),
  [SPOTIFY_ALBUM]: (state, action) => ({
    ...state,
    album: action.payload,
  }),
});

export default spotify;
