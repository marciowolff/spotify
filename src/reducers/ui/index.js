import createReducer from '../create-reducer';
import { ALERT_MESSAGE, LOADING } from './actions';

export const initialState = {
  alertMessage: '',
  loading: true,
};

const ui = createReducer(initialState, {
  [ALERT_MESSAGE]: (state, action) => ({
    ...state,
    alertMessage: action.payload.message,
  }),
  [LOADING]: (state, action) => ({
    ...state,
    loading: action.payload.loading,
  }),
});

export default ui;
