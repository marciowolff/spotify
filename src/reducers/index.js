import { combineReducers } from 'redux';
import ui from './ui';
import spotify from './spotify';

export default combineReducers({
  ui,
  spotify,
});
