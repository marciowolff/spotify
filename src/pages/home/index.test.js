import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { actionsSpotify } from '../../reducers/actions';

import Home from './index';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
  useSelector: () => ({ spotify: {} }),
}));

const mockSearch = jest.fn();
jest.mock('../../services', () => () => ({
  spotify: { search: mockSearch },
}));

const history = { push: jest.fn() };
const location = { pathname: '/teste' };
const spyactionsSpotify = jest.spyOn(actionsSpotify, 'setList');

let wrapper;

beforeEach(() => {
  mockSearch.mockImplementation(() =>
    Promise.resolve({ albums: { items: {} } })
  );
  wrapper = shallow(<Home location={location} history={history} />);
});

afterEach(() => {
  spyactionsSpotify.mockClear();
});

it('Thumb should be a function', () => {
  expect(typeof Home).toBe('function');
});

it('Should Search to match snapshot', () => {
  const tree = renderer
    .create(<Home location={location} history={history} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('render page default', () => {
  it('Should render component Container', () => {
    const CpContainer = wrapper.find('Container');
    expect(CpContainer).toHaveLength(1);
    expect(CpContainer.hasClass('page-home')).toBeTruthy();
  });

  it('Should render component Search', () => {
    const CpSearch = wrapper.find('Search');
    expect(CpSearch).toHaveLength(1);
    expect(CpSearch.props().id).toBe('search');
    expect(CpSearch.props().label).toBe(
      'Busque por artistas, álbuns ou músicas'
    );
    expect(CpSearch.props().onSearch).toBeTruthy();
  });

  it('Should render component Result', () => {
    const CpResult = wrapper.find('Result');
    expect(CpResult).toHaveLength(1);
    expect(CpResult.props().value).toBe('');
    expect(CpResult.props().data).toEqual({ spotify: {} });
    expect(CpResult.props().handleClick).toBeTruthy();
  });
});

describe('Events', () => {
  it('Should return onSearch', async () => {
    await wrapper
      .find('Search')
      .props()
      .onSearch('texto');

    expect(mockSearch).toHaveBeenCalledWith('texto');
    expect(spyactionsSpotify).toHaveBeenCalledWith({ items: {} });

    await wrapper
      .find('Search')
      .props()
      .onSearch('texto');

    expect(mockSearch.mock.calls.length).toBe(1);
    expect(spyactionsSpotify.mock.calls.length).toBe(1);
  });

  it('Should call handleClick Result redirect page albums', () => {
    wrapper
      .find('Result')
      .props()
      .handleClick({ id: 1 });

    expect(history.push).toHaveBeenCalledWith('album/1');
  });
});
