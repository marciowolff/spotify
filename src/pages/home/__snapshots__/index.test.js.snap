// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Search to match snapshot 1`] = `
Array [
  <header
    className="page-header"
  >
    <img
      id="logo"
      src="mock-file"
    />
  </header>,
  <main
    class="container page-home"
  >
    <div
      className="input-container "
    >
      <label
        className="label"
        htmlFor="search"
      >
        Busque por artistas, álbuns ou músicas
      </label>
      <input
        className="input"
        id="search"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyUp={[Function]}
        type="text"
      />
    </div>
    <div
      className="result"
    >
      <h1>
        Resultados encontrados para "
        
        "
      </h1>
      <ul />
    </div>
  </main>,
]
`;
